lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -511,6 +517,32 @@ instance FromText PoolId where
         where
             textDecodingError = Left . TextDecodingError . show
 
+-- | Coins are stored as Lovelace (reminder: 1 Lovelace = 1e-6 ADA)
+newtype Stake = Stake { getStake :: Word64 } deriving (Show, Eq, Generic)
+
+data StakeDistribution = StakeDistribution
+    { dangling :: Stake
+    , pools :: [(PoolId,Stake)]
+    , unassigned :: Stake
+    } deriving (Eq, Show, Generic)
```

Comment:
Agree for the types being here, yet, the `JSON` instances should not be here but in `Api.Types`, like the others :)

Reply:
used `ApiT` trick in `Cardano.Wallet.Jormungandr.Api`